File: helloworld.py
import time

word = 'hello world!'
newword = ''

for _ in word:
    newword += ' '

for i in range(len(word)):
    while word[i] != newword[i]:
        newword = newword[:i] + chr(ord(newword[i])+1) + newword[i+1:]
        print(newword)
        time.sleep(0.01)

File: helloworld.c
#include <stdio.h>
#include <string.h>
#include <windows.h>

void main() {
    char word[] = "Hello World!";
    char newword[] = "            ";

    for(int i = 0; i < strlen(word); i++) {
        char c = *(newword+i);
        while (*(word+i) != c) {
            printf("%s\n",newword);
            c++;
            newword[i] = c;
            Sleep(10);
        }
    }
    printf("%s",newword);
}